aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3308 @@
+
+DT_ARM_SYMTABSZ gives the number of entries in the dynamic symbol table,
+including the initial dummy symbol.
+
+DT_ARM_PREEMPTMAP holds the address of the pre-emption map for platforms that
+use the DLL static binding model.  See `Symbol Pre-emption in DLLs`_ for details.
+On platforms that permit use of a pre-emption map, the DT_SONAME tag must be
+present in all shared objects.
+
+.. note::
+   Some executable images may exist that use DT_ARM_RESERVED1 and
+   DT_ARM_RESERVED2 instead of DT_ARM_SYMTABSZ and DT_ARM_PREEMPTMAP
+   respectively.  These tags use the d_un field in a manner incompatible with
+   the Generic ELF requirements.
+
+Post-Link Processing
+--------------------
+
+For some execution environments a further processing step may be needed after
+linking before an executable can be run on the target environment.  The precise
+processing may depend on both the target platform.  Depending on the nature of
+the post-processing it may be done in any of following places
+
+* As a final step during linking
+
+* As a preliminary step during execution of the image
+
+* As a separate post-linking step
+
+In some cases the result may still be an ELF executable image, in others it may
+produce an image that is in some other format more appropriate to the operating
+system.
+
+Production of BE-8 images
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Images that are expected to execute in big-endian mode on processors that
+implement Architecture version 6 or higher will normally need to be
+post-processed to convert the instructions that are in big-endian byte order to
+little-endian as expected by the processor.  The mapping symbol information can
+be used to do this transformation accurately.  In all segments that contain
+executable code:
+
+* For areas mapped as data ($d or $d.<any…>) no changes are made
+
+* For areas mapped as Thumb ($t or $t.<any…>) each half-word aligned pair of
+  bytes are swapped
+
+* For areas mapped as Arm ($a or $a.<any…>) each word-aligned object is swapped
+  so that the first and fourth bytes are exchanged and the second and third
+  exchanged.
+
+An ELF image that has been transformed in this manner is marked by setting
+EF_ARM_BE8 in the e_flags field.
+
+.. note::
+   If BE-8 images are subject to further relocation of instructions (either by
+   a dynamic linker or by further post-linking operations) account must be
+   taken of the fact that the instructions are now in little-endian format.
+
+APPENDIX Specimen Code for PLT Sequences
+========================================
+
+The status of this appendix is informative.
+
+DLL-like, single address space, PLT linkage
+-------------------------------------------
+
+The simplest code sequence for the PLT entry corresponding to imported symbol X
+is:
+
+
+.. code-block::
+
+   LDR   ip, [pc, #0]         ; Load the 32-bit offset of my PLTGOT entry from SB
+   LDR   pc, [ip, sb]!        ; Branch indirect through the PLTGOT entry leaving ip addressing the PLTGOT slot
```

Comment:
I prefer to chop off these lines at 80, so the text doesn't get minified to fit on the line.